src/main/java/net/dv8tion/jda/internal/entities/ReceivedMessage.java

```diff
@@ -264,6 +239,7 @@ public ReactionPaginationAction retrieveReactionUsers(@Nonnull String unicode)
     @Override
     public MessageReaction.ReactionEmote getReactionById(@Nonnull String id)
     {
+        Checks.isSnowflake(id);
         return getReactionById(MiscUtil.parseSnowflake(id));
```

Comment:
This check is unnecessary cine `parseSnowflake` will throw if the input is an invalid snowflake anyway.

Reply:
I added the check as an easy way to consistently throw `IllegalArgumentExceptions` instead of throwing `NumberFormatExceptions` in some cases. https://github.com/DV8FromTheWorld/JDA/pull/1236#discussion_r402305343 is a relevant review comment to this.

Reply:
We consistenly do just parseSnowflake and rely on that throw. A `NumberFormatException` is a `IllegalArgumentException`.

https://docs.oracle.com/javase/8/docs/api/java/lang/NumberFormatException.html
> public class NumberFormatException
extends IllegalArgumentException